src/scaffoldmaker/meshtypes/meshtype_3d_bone1.py

```diff
@@ -0,0 +1,461 @@
+from scaffoldmaker.utils import vector
+from scaffoldmaker.utils.cylindermesh import CylinderMesh, CylinderShape, CylinderEnds, CylinderCentralPath
+from scaffoldmaker.utils.meshrefinement import MeshRefinement
+from scaffoldmaker.utils.shieldmesh import ShieldMesh3D
+from scaffoldmaker.utils.spheremesh import SphereMesh, SphereShape
+from scaffoldmaker.utils.zinc_utils import exnode_string_from_nodeset_field_parameters
+from scaffoldmaker.utils.derivativemoothing import DerivativeSmoothing
+from scaffoldmaker.annotation.annotationgroup import AnnotationGroup, findOrCreateAnnotationGroupForTerm
+
+
+class MeshType_3d_bone1 (Scaffold_base):
+    """
+Generates a solid cylinder using a ShieldMesh of all cube elements,
+with variable numbers of elements in major, minor, shell and axial directions.
+    """
+    centralPathDefaultScaffoldPackages = {
+        'Cylinder 1': ScaffoldPackage(MeshType_1d_path1, {
+            'scaffoldSettings': {
+                'Coordinate dimensions': 3,
+                'D2 derivatives': True,
+                'D3 derivatives': True,
+                'Length': 3.0,
+                'Number of elements': 3
+            },
+            'meshEdits': exnode_string_from_nodeset_field_parameters(
+                [Node.VALUE_LABEL_VALUE, Node.VALUE_LABEL_D_DS1, Node.VALUE_LABEL_D_DS2, Node.VALUE_LABEL_D2_DS1DS2,
+                 Node.VALUE_LABEL_D_DS3, Node.VALUE_LABEL_D2_DS1DS3], [
+                    (1, [[0.0, 0.0, 0.0], [0.0, 0.0, 1.0], [1.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 0.0]]),
+                    (2, [[0.0, 0.0, 1.0], [0.0, 0.0, 1.0], [1.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 0.0]]),
+                    (3, [[0.0, 0.0, 2.0], [0.0, 0.0, 1.0], [1.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 0.0]]),
+                    (4, [[0.0, 0.0, 3.0], [0.0, 0.0, 1.0], [1.0, 0.0, 0.0], [0.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 0.0]])
+                ])
+        })
+    }
+
+    @staticmethod
+    def getName():
+        return '3D Bone 1'
+
+    @classmethod
+    def getDefaultOptions(cls, parameterSetName='Default'):
+        centralPathOption = cls.centralPathDefaultScaffoldPackages['Cylinder 1']
+        options = {
+            'Central path': copy.deepcopy(centralPathOption),
+            'Number of elements across major': 6,
+            'Number of elements across minor': 6,
+            'Number of elements across shell': 1,
+            'Number of elements across transition': 1,
+            'Number of elements along': 4,
+            'Shell element thickness proportion': 1.0,
+            'Lower half': False,
+            'Use cross derivatives': False,
+            'Refine': False,
+            'Refine number of elements across major': 1,
+            'Refine number of elements along': 1,
+            'Upper scale': 1.0,
+            'Upper scale_Z': 1.0,
+            'Lower scale': 1.0,
+            'Lower scale_Z': 2.0
```

Comment:
Is 'Lower scale_Z' set as 2.0 for a specific reason?

Reply:
Changed to 1